File: sample/CStyleSamplesCS/Samples/Image2Stream.cs
using System;
using System.IO;
using OpenCvSharp;
using SampleBase;

namespace CStyleSamplesCS
{
    /// <summary>
    /// 
    /// </summary>
    class Image2Stream
    {
        public Image2Stream()
        {
            // Stream -> IplImage
            using (FileStream stream = new FileStream(FilePath.Image.Lenna, FileMode.Open))
            using (IplImage img = IplImage.FromStream(stream, LoadMode.Color))
            {
                CvWindow.ShowImages(img);

                // IplImage -> Stream
                using (MemoryStream ms = new MemoryStream())
                {
                    img.ToStream(ms, ".tiff");
                    ms.ToString();
                }
            }

            // Stream -> CvMat
            using (FileStream stream = new FileStream(FilePath.Image.Lenna, FileMode.Open))
            using (CvMat mat = CvMat.FromStream(stream, LoadMode.Color))
            {
                mat.ToString();

                // CvMat -> Stream
                using (MemoryStream ms = new MemoryStream())
                {
                    mat.ToStream(ms, ".bmp");
                    ms.ToString();
                }
            }
        }
    }
}

File: sample/CStyleSamplesVB/Samples/Image2Stream.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text
Imports OpenCvSharp

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' 
''' </summary>
Friend Module Image2Stream
    Public Sub Start()
        ' Stream -> IplImage
        Using stream As New FileStream(FilePath.Image.Lenna, FileMode.Open)
            Using img As IplImage = IplImage.FromStream(stream, LoadMode.Color)
                CvWindow.ShowImages(img)

                ' IplImage -> Stream
                Using ms As New MemoryStream()
                    img.ToStream(ms, ".tiff")
                    ms.ToString()
                End Using
            End Using
        End Using

        ' Stream -> CvMat
        Using stream As New FileStream(FilePath.Image.Lenna, FileMode.Open)
            Using mat As CvMat = CvMat.FromStream(stream, LoadMode.Color)
                mat.ToString()

                ' CvMat -> Stream
                Using ms As New MemoryStream()
                    mat.ToStream(ms, ".bmp")
                    ms.ToString()
                End Using
            End Using
        End Using
    End Sub
End Module
' End Namespace
